improver/calibration/reliability_calibration.py

```diff
@@ -488,6 +488,289 @@ def process(self, cubes, coordinates=None):
+        bin to the highest probability bin. Compare each pair of bins and, if
+        a pair is non-monotonic, replace the value of the upper bin with the
+        value of the lower bin.
+
+        Args:
+            observation_count (numpy.ndarray):
+                Observation count extracted from reliability table.
+            forecast_count (numpy.ndarray):
+                Forecast count extracted from reliability table.
+
+        Returns:
+            numpy.ndarray:
+                Observation count computed from a monotonic observation frequency.
+
+        """
+        observation_frequency = np.array(observation_count / forecast_count)
+        for index, lower_bin in enumerate(observation_frequency[:-1]):
+            (diff,) = np.diff([lower_bin, observation_frequency[index + 1]])
+            if diff < 0:
+                observation_frequency[index + 1] = lower_bin
+        observation_count = observation_frequency * forecast_count
+        return observation_count
+
+    @staticmethod
+    def _update_reliability_table(
+        reliability_table,
+        observation_count,
+        forecast_probability_sum,
+        forecast_count,
+        probability_bin_coord,
+    ):
+        """
+        Update the reliability table data and the probability bin coordinate.
+
+        Args:
+            reliability_table (iris.cube.Cube):
+                A reliability table to be manipulated.
+            observation_count (numpy.ndarray):
+                Observation count extracted from reliability table.
+            forecast_probability_sum (numpy.ndarray):
+                Forecast probability sum extracted from reliability table.
+            forecast_count (numpy.ndarray):
+                Forecast count extracted from reliability table.
+            probability_bin_coord (iris.coord.DimCoord):
+                Original probability bin coordinate.
+
+        Returns:
+            iris.cube.Cube
+                Updated reliability table.
+        """
+        if len(probability_bin_coord.points) < len(
+            reliability_table.coord(probability_bin_coord).points
+        ):
+            reliability_table = reliability_table[..., 0:-1]
+            reliability_table.replace_coord(probability_bin_coord)
+        obs_count_cube = reliability_table.extract(
+            iris.Constraint(table_row_name="observation_count")
+        )
+        obs_count_cube.data = observation_count
+        forecast_count_cube = reliability_table.extract(
+            iris.Constraint(table_row_name="forecast_count")
+        )
+        forecast_count_cube.data = forecast_count
+        forecast_prob_sum_cube = reliability_table.extract(
+            iris.Constraint(table_row_name="sum_of_forecast_probabilities")
+        )
+        forecast_prob_sum_cube.data = forecast_probability_sum
+        reliability_table = iris.cube.CubeList(
+            [obs_count_cube, forecast_prob_sum_cube, forecast_count_cube]
+        ).merge_cube()
```

Comment:
If you were still looking for a simpler way of doing this, the following would work.

```suggestion
        final_data = np.stack(
            [observation_count, forecast_probability_sum, forecast_count]
        )
        nrows, ncols = final_data.shape
        reliability_table = reliability_table[0:nrows, 0:ncols].copy(data=final_data)
        reliability_table.replace_coord(probability_bin_coord)
```

Reply:
I guess this is assuming we always have the coordinate in that order. Is that guaranteed?

Reply:
If you check out line 82 in this file you will see that they are set into that order.

Reply:
Ok, done